experimental/packages/otlp-transformer/src/metrics/internal.ts

```diff
@@ -13,125 +13,126 @@
-  } else if (metric.aggregator.kind === AggregatorKind.LAST_VALUE) {
-    out.gauge = toGauge(metric, startTime);
-  } else if (metric.aggregator.kind === AggregatorKind.HISTOGRAM) {
-    out.histogram = toHistogram(metric, startTime);
+  const aggregationTemporality = toAggregationTemporality(metricTemporality);
+
+  if (metricData.dataPointType === DataPointType.SINGULAR) {
+    const dataPoints = toSingularDataPoints(metricData);
+    const isMonotonic = metricData.descriptor.type === InstrumentType.COUNTER ||
+      metricData.descriptor.type === InstrumentType.OBSERVABLE_COUNTER;
+    if (isSum(metricData)) {
+      out.sum = {
+        aggregationTemporality,
+        isMonotonic,
+        dataPoints
+      };
+
+    } else {
+      // Instrument is a gauge.
+      out.gauge = {
+        dataPoints
+      };
+    }
+  } else if (isHistogram(metricData)) {
+    out.histogram = {
+      aggregationTemporality,
+      dataPoints: toHistogramDataPoints(metricData)
+    };
   }
 
   return out;
 }
 
-function isSum(metric: MetricRecord) {
-  return metric.aggregator.kind === AggregatorKind.SUM ||
-    metric.descriptor.metricKind === MetricKind.OBSERVABLE_COUNTER ||
-    metric.descriptor.metricKind === MetricKind.OBSERVABLE_UP_DOWN_COUNTER;
-}
-
-function toAggregationTemporality(
-  metric: MetricRecord
-): EAggregationTemporality {
-  if (metric.descriptor.metricKind === MetricKind.OBSERVABLE_GAUGE) {
-    return EAggregationTemporality.AGGREGATION_TEMPORALITY_UNSPECIFIED;
-  }
+function toSingularDataPoint(dataPoint: DataPoint<number> | DataPoint<Histogram>, valueType: ValueType) {
+  const out: INumberDataPoint = {
+    attributes: toAttributes(dataPoint.attributes),
+    startTimeUnixNano: hrTimeToNanoseconds(
+      dataPoint.startTime
+    ),
+    timeUnixNano: hrTimeToNanoseconds(
+      dataPoint.endTime
+    ),
+  };
 
-  if (metric.aggregationTemporality === AggregationTemporality.AGGREGATION_TEMPORALITY_DELTA) {
-    return EAggregationTemporality.AGGREGATION_TEMPORALITY_DELTA;
+  if (valueType === ValueType.INT) {
+    out.asInt = dataPoint.value as number;
   }
 
-  if (metric.aggregationTemporality === AggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE) {
-    return EAggregationTemporality.AGGREGATION_TEMPORALITY_CUMULATIVE;
+  if (valueType === ValueType.DOUBLE) {
+    out.asDouble = dataPoint.value as number;
   }
 
-  return EAggregationTemporality.AGGREGATION_TEMPORALITY_UNSPECIFIED;
+  return out;
 }
 
-function toSum(
-  metric: MetricRecord,
-  startTime: number,
-): ISum {
-  return {
-    dataPoints: [toNumberDataPoint(metric, startTime)],
-    isMonotonic:
-      metric.descriptor.metricKind === MetricKind.COUNTER ||
-      metric.descriptor.metricKind === MetricKind.OBSERVABLE_COUNTER,
-    aggregationTemporality: toAggregationTemporality(metric),
-  };
+function toSingularDataPoints(
+  metricData: MetricData
+): INumberDataPoint[] {
+  return Array.from(metricData.dataPoints.map(dataPoint => {
```

Comment:
Is the `Array.from` needed if it's calling map or was the intention here `Array.from(metricData.dataPoints, dataPoint => ...`?

Reply:
True, this is not necessary. Removed it in [dea3327](https://github.com/open-telemetry/opentelemetry-js/pull/2871/commits/dea3327932a044f2ab1594bdf61904c61762b06b).
Thanks! :slightly_smiling_face: